tests/integration.test.js

```diff
@@ -0,0 +1,254 @@
+const supertest = require('supertest');
```

Comment:
In a future PR, we should really split these tests into separate files. Fine for now though, I can do that (I want to write a bunch of tests some time soon anyway)